File: Easy leetcode/2011. Final Value of Variable After Performing Operations.cpp

class Solution {
public:
    int finalValueAfterOperations(vector<string>& operations) {
        int x = 0; // Initial value of X
        for (string op : operations) { // Iterate through each operation
            if (op == "++X" || op == "X++") {
                x++; // Increment X by 1
            } else if (op == "--X" || op == "X--") {
                x--; // Decrement X by 1
            }
        }
        return x; // Return the final value of X
    }
};

File: Easy leetcode/2011. Final Value of Variable After Performing Operations.py
class Solution:
    def finalValueAfterOperations(self, operations: list[str]) -> int:
        x = 0  # Initial value of X
        for op in operations:  # Iterate through each operation
            if op in ("++X", "X++"):
                x += 1  # Increment X by 1
            elif op in ("--X", "X--"):
                x -= 1  # Decrement X by 1
        return x  # Return the final value of X
